code/online_challenges/project_euler/problem_036/problem_036.cpp

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <string>
+
+std::string convertBase2(int number);
+bool isPalindrome(const std::string& str);
+
+int main()
+{
+    int sum = 0;
+    
+    for (int i = 1; i < 1000000; ++i)
+    {
+        if (isPalindrome(std::to_string(i)) && isPalindrome(convertBase2(i)))
+            sum += i;
+    }
+    std::cout << sum << "\n";
+}
+
+std::string convertBase2(int number)
```

Comment:
maybe this can use [bitset](http://en.cppreference.com/w/cpp/utility/bitset/to_string)

Reply:
I'll do that.

Reply:
Ok, @iattempt I did this, but the code was slower when I did so with optimizations. My current code ran in ~235ms, while the bitset implementation took around ~350ms. Thus, I will keep the current implementation.

Reply:
Here is the bitset code that I used for benchmarking:

```C++
#include <bitset>
#include <iostream>
#include <string>

std::string convertBase2(int number);
bool isPalindrome(const std::string& str);

int main()
{
    int sum = 0;
    
    for (int i = 1; i < 1000000; ++i)
    {
        std::string currentBinaryString = std::bitset<32>(i).to_string();
        currentBinaryString.erase(0, currentBinaryString.find('1')); // Remove leading zeroes
        if (isPalindrome(std::to_string(i)) && isPalindrome(currentBinaryString))
            sum += i;
    }
    std::cout << sum << "\n";
}

bool isPalindrome(const std::string& str)
{
    return str == std::string{ str.rbegin(), str.rend() };
}
```